### PR TITLE
Fix typo and markdown format error

### PR DESCRIPTION
**Pull request by hainesc**
I am reading the docs and find some typos, please check the patch, thanks.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/9943)

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -315,7 +315,7 @@ encounter data that necessitate conflict resolution
 
 When a transaction restarts, it changes its priority and/or moves its
 timestamp forward depending on data tied to the conflict, and
-begins anew reusing the same txn id. The prior run of the transaction might
+begins a new reusing the same txn id. The prior run of the transaction might
```
Comment (tamird) [inline docs/design.md]: "anew" is correct.

**Comment by tamird** [comment]
LGTM except for one comment

File: docs/design.md (hunk 1 above)
Comment (hainesc) [inline docs/design.md]: Sorry for the delay, I will correct it. Thanks.

**Comment by hainesc** [comment]
I have reverted the code and rebased to one commit.

**Comment by tamird** [comment]
Thank you!

**Pull request merged**